documentation/docs/30-advanced/25-errors.md

```diff
@@ -77,11 +77,11 @@ By default, unexpected errors are printed to the console (or, in production, you
 { "message": "Internal Error" }
 ```
 
-Unexpected errors will go through the [`handleError`](/docs/hooks#shared-hooks-handleerror) hook, where you can add your own error handling — for example, sending errors to a reporting service, or returning a custom error object. 
+Unexpected errors will go through the [`handleError`](/docs/hooks#shared-hooks-handleerror) hook, where you can add your own error handling — for example, sending errors to a reporting service, or returning a custom error object.
 
 ```js
 /// file: src/hooks.server.js
-// @errors: 2322 2571
+// @errors: 2322 2571 18046 2339
```

Comment:
we probably don't need 18046 and 2339. I think adding 18046 is probably better than using `?.` as I don't think error should really be `undefined`

Reply:
it's `unknown`, which I think is as it should be

Reply:
I was thinking the user would define the type of `error` using `src/app.d.ts`: https://kit.svelte.dev/docs/errors#type-safety

If the user does that then I think the type would no longer be `unknown` and you don't need the `?.` or 2339

If we do want to keep the `?.` then I think you could probably remove 18046 as it's not included above

Reply:
`App.Error` is the _output_ of `handleError`, not the input. It's `unknown` because TypeScript has no real way of knowing that some library code somewhere doesn't do `throw 'potato'`